### DISCUSSION TITLE
[Feature Request] Carousel component

### DISCUSSION THREAD

**Discussion by srivatsa17** [heroui-inc/heroui#4678] [Feature requests]
### Is your feature request related to a problem? Please describe.

I am trying to build an ecommerce website where i want to display a slideshow of images but unable to do so because there is no carousel component available yet.

### Describe the solution you'd like

It would be great to have a carousel for slideshow of images/text. 
Ref: https://react-bootstrap.github.io/docs/components/carousel

### Describe alternatives you've considered

Would be displaying a single image instead of slide show.

**Comment by DhruvParikh1** [comment]
This would be extremely convenient. I have been using `react-slick` while I wait for this to be added. Probably a core component to any UI library

**Comment by krishnapraveen7** [comment]
Is this on roadmap?

**Comment by artiphishle** [comment]
Feature proposal:

- swipe on mobile device
- optional loop
- optional pagination
- optional title/text on image
- next/prev arrows
- keyboard navigation
- slide stops on mouseover image
- event callbacks
- slide duration

Event callbacks also pretty standard, I e.g. would like to change theme (next-themes) on slide ;)

in a version 2 it would be nice to add multiple effects for slides 🤩 Maybe by just supporting some known plugins, there are libraries out there to integrate if this is ok policy-wise, as it would push some 3rd party libraries

and 3d carousel (not too hard, one Math.tan() calculation and a perspective 🤗 at least that's how I did it back in the days)

I need to find out how easy it is to collaborate, e.g. what kind of mobile event handlers/package us used etc.

If it's easy I might inplement it here.

Let me know anything that helps to quickly collaborate, thanks in advance, going to check it out too or start doing it standalone if not easy to integrate, as I'm a bit in a hurry on this ;)

A more general thing for every nextUI component could be (if not so already) to simply add items by mentioning a API route (or server action) as such an abstraction would tidy up client-side code a lot, but that's offtopic here, aplogies 🐱

**Comment by 0xc0000f** [comment]
++

**Comment by theteleporter** [comment]
It would be nice if it was integrated with [embla carousel](https://www.embla-carousel.com) too, it has really nice features, such as the parallax, it's really nice.

**Comment by zvictor** [comment]
I am in need of a carousel atm as well, but I wonder if that should be NextUI's job to include one.
With so many options out there (thanks @theteleporter for the suggestion, I didn't know embla!), isn't it better to keep that as a separate concern? Honestly asking!

Maybe we should instead be discussing what external tool best fit the needs of our community.
Unless I am missing the point here, of course 😊

**Comment by theteleporter** [comment]
> I am in need of a carousel atm as well, but I wonder if that should be NextUI's job to include one.
> With so many options out there (thanks @theteleporter for the suggestion, I didn't know embla!), isn't it better to keep that as a separate concern? Honestly asking!
> 
> Maybe we should instead be discussing what external tool best fit the needs of our community.
> Unless I am missing the point here, of course 😊

You can check out [Shadcn UI's Carousel](https://ui.shadcn.com/docs/components/carousel), they have simplified the [Embla Carousel](https://embla-carousel.com) and Integrated it with their components. 
I think Next UI should also follow what Shadcn are doing.

**Comment by Ayan-Ali** [comment]
+1

**Comment by nyakaz73** [comment]
+1

**Comment by hhniao** [comment]
> This would be extremely convenient. I have been using `react-slick` while I wait for this to be added. Probably a core component to any UI library

@DhruvParikh1 there have some bug with `react-slick`

how can you fix it ?

[https://codesandbox.io](https://codesandbox.io/p/devbox/new-night-8nwzdf)

the docs label on the codesandbox

**Comment by justincorporated** [comment]
+1, we absolutely NEED this in the library ASAP — this seems like a must-have component that would play very well alongside the 'Testimonials Scrolling Banner' component in the Pro library

**Comment by DhruvParikh1** [comment]
> > This would be extremely convenient. I have been using `react-slick` while I wait for this to be added. Probably a core component to any UI library
> 
> 
> 
> @DhruvParikh1 there have some bug with `react-slick`
> 
> 
> 
> how can you fix it ?
> 
> 
> 
> [https://codesandbox.io](https://codesandbox.io/p/devbox/new-night-8nwzdf)
> 
> 
> 
> the docs label on the codesandbox

What and where is the bug?

**Comment by Tarkan-Zarrouk** [comment]
+1
this would be awesome

**Comment by Rain-YuXia** [comment]
+1
Carousel would be a must-have component in any other UI library

**Comment by itorz7** [comment]
+1

**Comment by maryam-zafarr** [comment]
+1

**Comment by AriaFantom** [comment]
+1

**Comment by ScottAgirs** [comment]
+1

**Comment by Denys1209** [comment]
+1

**Comment by leosantacruz** [comment]
I've been using Keen-slider for a while but I would love to see a slider on NextUI!

**Comment by casually-creative** [comment]
For reference and inspiration, NuxtUI has a carousel: https://ui.nuxt.com/components/carousel.

**Comment by leotaozeng** [comment]
+1 would be really nice to have this component added

**Comment by Softjey** [comment]
+1

**Comment by nextrie** [comment]
+1

**Comment by elkira01** [comment]
+1 please

**Comment by sotnii** [comment]
+1

**Comment by lfraggioli-vulcan** [comment]
+1

**Comment by galkin** [comment]
+1

**Comment by AhmedSamy9999** [comment]
+1

**Comment by henry-oway** [comment]
+1

**Comment by xxmistacruzxx** [comment]
+1

**Comment by UnoMaestro** [comment]
+1

**Comment by RazerGhost** [comment]
+1

**Comment by rayaqin** [comment]
if I had known this is missing from this library we wouldn't have chosen heroui, how the hell can you not add carousel to a component lib

**Reply by elkira01** [reply]
I feel you; but you would be surprised to find out how many quite good popular UI lib out there doesn't provide a carousel either

**Comment by melodyxpot** [comment]
+1

**Comment by artiphishle** [comment]
https://github.com/user-attachments/assets/8690a966-7045-4de6-9465-d122d233675b

I've started a contribution today. Will keep you up-to-date ;)

**Comment by artiphishle** [comment]
https://github.com/user-attachments/assets/27592f44-d501-4f82-98ac-4960a0721788

(Had to compress the video to upload, so no quality here)
This is just the very beginning...

In general you can add whatever children you like to the carousel, here I'm using HeroUI Card components.

👍 I'm very happy that the repo is very intuitive to start working with.

**Comment by erikdemarco** [comment]
It's been 3 years and still no sign of updates. Can we safely say carousel will never be implemented?

**Reply by cyberxnomad** [reply]
This feature is on the roadmap of v3.

**Comment by TheCodister** [comment]
This would be extremely convenient. I have already used HeroUI back in 2024 and I have to installed another library in order to implement the carousel.
